lib/predicates/unique.rb

```diff
@@ -41,16 +41,24 @@ def validate(value, record)
 
     !klass.where(sql.join(" AND ")).exists?
   end
-  
+
   protected
-  
+
   def comparison_for(field, value, klass)
     quoted_field = "#{klass.quoted_table_name}.#{klass.connection.quote_column_name(field)}"
 
     if klass.columns_hash[field.to_s].text?
       if case_sensitive
         # case sensitive text comparison in any database
-        klass.send(:sanitize_sql, ["#{quoted_field} #{klass.connection.case_sensitive_equality_operator} ?", value])
+        if klass.connection.respond_to?(:case_sensitive_equality_operator)
+          # <= Rails 4.1
+          comparison = klass.send(:sanitize_sql, ["#{quoted_field} #{klass.connection.case_sensitive_equality_operator} ?", value])
+        else
+          # >= Rails 4.2
+          case_sensitive_value = klass.connection.case_sensitive_modifier(value, field)
+          case_sensitive_value = case_sensitive_value.to_sql if case_sensitive_value.respond_to?(:to_sql)
+          klass.send(:sanitize_sql, ["#{quoted_field} = ?", case_sensitive_value])
```

Comment:
apologies, but hardcoding the `=` was the easiest way to fix this. Otherwise we have to get AREL involved, which is a bigger change. I'd say people with database that have other operators can push PRs if they want this.